lib/internal/backend/repository/repository2.go

```diff
@@ -167,48 +175,146 @@ func (rb *RepositoryBackend) buildACIV2(layerIDs []string, dockerURL *types.Pars
 	return aciLayerPaths, aciManifests, nil
 }
 
-func (rb *RepositoryBackend) getManifestV2(dockerURL *types.ParsedDockerURL) (*v2Manifest, []string, error) {
+func (rb *RepositoryBackend) buildACIV22(layerIDs []string, dockerURL *types.ParsedDockerURL, outputDir string, tmpBaseDir string, compression common.Compression) ([]string, []*schema.ImageManifest, error) {
+	layerFiles := make([]*os.File, len(layerIDs))
+
+	tmpParentDir, err := ioutil.TempDir(tmpBaseDir, "docker2aci-")
+	if err != nil {
+		return nil, nil, err
+	}
+	defer os.RemoveAll(tmpParentDir)
+
+	copier := progressutil.NewCopyProgressPrinter()
+
+	var errChannels []chan error
+	closers := make([]io.ReadCloser, len(layerIDs))
+	for i, layerID := range layerIDs {
+		errChan := make(chan error, 1)
+		errChannels = append(errChannels, errChan)
+		// https://github.com/golang/go/wiki/CommonMistakes
+		i := i // golang--
+		layerID := layerID
+		go func() {
+			tmpDir, err := ioutil.TempDir(tmpParentDir, "")
+			if err != nil {
+				errChan <- fmt.Errorf("error creating dir: %v", err)
+				return
+			}
+
+			layerFiles[i], closers[i], err = rb.getLayerV2(layerID, dockerURL, tmpDir, copier)
+			if err != nil {
+				errChan <- fmt.Errorf("error getting the remote layer: %v", err)
```

Comment:
you are using `errChan` as the sync mechanism to propagate changes of `layerFiles[i]`, and `closers[i]`, this is pretty fragile. Wouldn't it be easier to simply send a `layer struct` via a channel it being:
```
type layer struct {
  file *os.File
  c io.ReadCloser
  err error
}
```